src/components/tasks/task-dialog.tsx

```diff
@@ -506,14 +754,220 @@ const TaskDialog = ({ projectId, milestoneId, open, task, onClose }: Props) => {
             </TableBody>
           </Table>
         </TableContainer>
-        <Button variant="text" color="primary" sx={{ borderRadius: 25 }} onClick={() => setAttachmentDialogOpen(true)}>
-          <AddIcon />
-          {t("newMilestoneTaskDialog.taskAttachmentsTable.addButton")}
-        </Button>
+        <div style={{ display: "flex", justifyContent: "flex-end" }}>
+          <Button
+            variant="text"
+            color="primary"
+            sx={{ borderRadius: 25 }}
+            onClick={() => setAttachmentDialogOpen(true)}
+          >
+            <AddIcon />
+            {t("newMilestoneTaskDialog.taskAttachmentsTable.addButton")}
+          </Button>
+        </div>
       </>
     );
   };
 
+  /**
+   * Renders a change proposal edit form
+   *
+   * @param changeProposal change proposal
+   */
+  const renderExistingChangeProposals = (changeProposal: ChangeProposal) => {
+    if (!changeProposal?.id) return;
+
+    const formattedStartDate = changeProposal.startDate ? DateTime.fromJSDate(changeProposal.startDate) : null;
+    const formattedEndDate = changeProposal.endDate ? DateTime.fromJSDate(changeProposal.endDate) : null;
+
+    const statusToLocalizedString = {
+      [ChangeProposalStatus.Approved]: t("changeProposals.accepted"),
+      [ChangeProposalStatus.Pending]: t("changeProposals.waitingForApproval"),
+      [ChangeProposalStatus.Rejected]: t("changeProposals.abandoned"),
+    };
+    const label = statusToLocalizedString[changeProposal.status];
```

Comment:
It is unnecessary to repeat the translate (t) function here, as it could be called wherever the localization key is used. Also, you can directly return the 'label' with
```
    const label = {
      [ChangeProposalStatus.Approved]: "changeProposals.accepted",
      [ChangeProposalStatus.Pending]: "changeProposals.waitingForApproval",
      [ChangeProposalStatus.Rejected]: "changeProposals.abandoned",
    }[changeProposal.status];
```